### PR TITLE
feat(EMS-1993): Policy and exports - Name on policy page

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Introduction
* Needed to add Name on policy page so exporter can choose if they are the name on the policy or someone else is

# Resolution
* Added Name on policy page, routes, controller and content strings
* Added validation
* Added tests
* Added e2e tests

* Note - DB change needed for displaying exporters name on radio which will be done in a future PR

### CODE REVIEW

File: e2e-tests/commands/insurance/complete-and-submit-name-on-policy-form.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { FIELD_IDS } from '../../constants';
```
Comment (ttbarnes) [inline e2e-tests/commands/insurance/complete-and-submit-name-on-policy-form.js:1]: Could we do `import INSURANCE_FIELD_IDS from '../../constants/field-ids/insurance';`, save some destructuring below?

File: e2e-tests/commands/insurance/complete-and-submit-name-on-policy-form.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      NAME_ON_POLICY: {
+        POSITION,
+        SAME_NAME,
+        OTHER_NAME,
+      },
+    },
+  },
+} = FIELD_IDS;
+
+const completeAndSubmitNameOnPolicyForm = ({ sameName = false }) => {
```
Comment (ttbarnes) [inline e2e-tests/commands/insurance/complete-and-submit-name-on-policy-form.js:16]: documentation

File: e2e-tests/constants/field-ids/insurance/policy-and-exports/index.js [hunk 3]
```diff
@@ -27,4 +27,10 @@ export const POLICY_AND_EXPORTS = {
     DESCRIPTION: 'goodsOrServicesDescription',
     FINAL_DESTINATION: 'finalDestinationCountryCode',
   },
+  NAME_ON_POLICY: {
+    NAME: 'nameOnPolicy',
```
[the remark this answers is not in the document]
Comment (ttbarnes) [reply e2e-tests/constants/field-ids/insurance/policy-and-exports/index.js:31]: Sorry ignore just seen changes below.

File: e2e-tests/insurance/cypress/e2e/journeys/policy-and-exports/name-on-policy/name-on-policy-page.spec.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+const {
+  INSURANCE: {
+    ROOT: INSURANCE_ROOT,
+    POLICY_AND_EXPORTS: {
+      ABOUT_GOODS_OR_SERVICES,
+      CHECK_YOUR_ANSWERS,
+      NAME_ON_POLICY,
+    },
+  },
+} = ROUTES;
```
Comment (ttbarnes) [inline e2e-tests/insurance/cypress/e2e/journeys/policy-and-exports/name-on-policy/name-on-policy-page.spec.js:27]: `INSURANCE_ROUTES`?

File: e2e-tests/insurance/cypress/e2e/journeys/policy-and-exports/name-on-policy/name-on-policy-page.spec.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    });
+  });
+
+  describe('form submission', () => {
+    it(`should redirect to ${CHECK_YOUR_ANSWERS}`, () => {
+      cy.navigateToUrl(url);
+
+      cy.completeAndSubmitNameOnPolicyForm({ sameName: true });
+
+      const expectedUrl = `${Cypress.config('baseUrl')}${INSURANCE_ROOT}/${referenceNumber}${CHECK_YOUR_ANSWERS}`;
```
Comment (ttbarnes) [inline e2e-tests/insurance/cypress/e2e/journeys/policy-and-exports/name-on-policy/name-on-policy-page.spec.js:119]: Can we create a `baseUrl` const at the top of the file to be DRY, started doing this in other tests

File: src/ui/server/controllers/insurance/policy-and-export/name-on-policy/validation/rules/position.test.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+      };
+
+      const result = positionRule(mockSubmittedData, mockErrors);
+
+      expect(result).toEqual(mockErrors);
+    });
+  });
+
+  describe(`when the field is not provided and ${SAME_NAME} is selected`, () => {
+    it('should return validation error', () => {
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/policy-and-export/name-on-policy/validation/rules/position.test.ts:49]: Could be simplified to test it's returning the result of `emptyFieldValidation`

File: src/ui/server/controllers/insurance/policy-and-export/type-of-policy/index.test.ts [hunk 7]
```diff
@@ -204,6 +206,7 @@ describe('controllers/insurance/policy-and-export/type-of-policy', () => {
     describe('api error handling', () => {
       beforeEach(() => {
         req.body = {
+          // @ts-ignore
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/policy-and-export/type-of-policy/index.test.ts:209]: Why do we have the ts-ignores?

Comment (Zainzzkk) [reply src/ui/server/controllers/insurance/policy-and-export/type-of-policy/index.test.ts:209]: With the things I have added it was causing a linting error in the tests and could not figure out why

File: src/ui/server/controllers/insurance/policy-and-export/name-on-policy/index.test.ts [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+      expect(res.render).toHaveBeenCalledWith(TEMPLATE, expectedVariables);
+    });
+
+    describe('when there is no application', () => {
+      beforeEach(() => {
+        res.locals = mockRes().locals;
+      });
+
+      it(`should redirect to ${PROBLEM_WITH_SERVICE}`, async () => {
+        await get(req, res);
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/policy-and-export/name-on-policy/index.test.ts:107]: No need for `await get` in any of the tests, the controller still has `async` as well, we don't need it - jest probably throwing a warning

**Review by ttbarnes — APPROVED** [review]

**Pull request merged**